language/expressions.md

```diff
@@ -720,9 +726,26 @@ result. Chained assignments are permitted.
 
 Examples:
 
+~~~puppet
+
     $a = 10
     $x = $y = 0
 
+    # from array
+    [$a, $b] = [1, 2]  # assigns 1 to $a, and 2 to $b
+
+    # from hash
+    [$a, $b] = { a => 10, b => 20, c => 30 }  # assigns 10 to $a, and 20 to $b
+
+    # from class
+    class mymodule::someclass::example($x = 100) {
+      $a = 10
+    }
+    include example
+    [$a, $x] = Class['mymodule::someclass::example']  # assigns 10 to $a, and 100 to $x
```

Comment:
A class added to the catalog using `class { classname: }` is lazily evaluated. All other - via include, are available immediately. There is really no such thing as a "private" variable in a class (except those that start with an underscore - since they are not valid in fully qualified name form. 
Your example works because both your instantiation of "meh" and the "notice" are lazily evaluated (and the notice is behind the "meh" in the queue...